### PR TITLE
implemented deposit event emitter functions

### PR DESCRIPTION
**Pull request by sparqet**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the current behavior?

Deposit event emitters missing

Resolves: #NA

## What is the new behavior?

- Added Deposit event emitters

## Does this introduce a breaking change?

 No

### CODE REVIEW

**Review by Sk8erboi84 — COMMENTED** [review]
Can you please add deposit in the book

File: src/event/event_emitter.cairo [hunk 1]
```diff
@@ -375,5 +435,67 @@ mod EventEmitter {
...
+            let initial_short_token_amount = deposit.initial_short_token_amount;
+            let min_market_tokens = deposit.min_market_tokens;
+            let updated_at_block = deposit.updated_at_block;
+            let execution_fee = deposit.execution_fee;
+            let callback_gas_limit = deposit.callback_gas_limit;
+            self
+                .emit(
+                    DepositCreated {
+                        key,
+                        account,
```
Comment (zarboq) [inline src/event/event_emitter.cairo:460]: You can do account: deposit.account

Same for other fields

**Review by zarboq — COMMENTED** [review]
test if possible

File: src/event/event_emitter.cairo [hunk 2]
```diff
@@ -107,6 +109,24 @@ trait IEventEmitter<TContractState> {
...
+    #[inline(always)]
+    fn emit_deposit_created(ref self: TContractState, key: felt252, deposit: Deposit,);
+
+    /// Emits the `DepositExecuted` event.
+    fn emit_deposit_executed(
+        ref self: TContractState,
+        key: felt252,
+        long_token_amount: u128,
+        short_token_amount: u128,
+        received_market_tokens: u128,
```
Comment (Sk8erboi84) [inline src/event/event_emitter.cairo:123]: no point keeping the last comma

File: src/event/event_emitter.cairo [hunk 3]
```diff
@@ -107,6 +109,24 @@ trait IEventEmitter<TContractState> {
...
+    );
+
+    /// Emits the `DepositCancelled` event.
+    fn emit_deposit_cancelled(
+        ref self: TContractState, key: felt252, reason: felt252, reasonBytes: Array<felt252>,
```
Comment (Sk8erboi84) [inline src/event/event_emitter.cairo:128]: no point keeping the last comma

File: src/event/event_emitter.cairo [hunk 4]
```diff
@@ -375,5 +435,53 @@ mod EventEmitter {
...
+                        initial_long_token: deposit.initial_long_token,
+                        initial_short_token: deposit.initial_short_token,
+                        long_token_swap_path: deposit.long_token_swap_path,
+                        short_token_swap_path: deposit.short_token_swap_path,
+                        initial_long_token_amount: deposit.initial_long_token_amount,
+                        initial_short_token_amount: deposit.initial_short_token_amount,
+                        min_market_tokens: deposit.min_market_tokens,
+                        updated_at_block: deposit.updated_at_block,
+                        execution_fee: deposit.execution_fee,
+                        callback_gas_limit: deposit.callback_gas_limit,
```
Comment (Sk8erboi84) [inline src/event/event_emitter.cairo:459]: no point keeping the last comma

**Pull request merged**